union-type.js

```diff
@@ -127,4 +127,22 @@ function Type(desc) {
 
 Type.check = true;
 
+Type.ListOf = function (T) {
+  var List = Type({List:[Array]});
+  var innerType = Type({T: [T]}).T;
+  var validate = List.case({
+    List: function (array) {
+      try{
+        for(var n in array) {
```

Comment:
i thought `for in` is standard from ES1
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in